### PR TITLE
feat: allow to reset config once dialog has been opened

### PR DESCRIPTION
**Pull request by va-stefanek**
## PR Checklist

Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: CONTRIBUTING.md#commit
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type

What kind of change does this PR introduce?

```
[ ] Bugfix
[x] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[ ] Documentation content changes
[ ] Other... Please describe:
```

## What is the current behavior?

We are not allowed to override config once it has been opened

Issue Number: #41 

## What is the new behavior?

We can override config of the dialog after opening

### CODE REVIEW

**Comment by va-stefanek** [comment]
@NetanelBasal do you have any idea how I can write tests for it? I see that we are not testing dialogRef methods like onClose, onReset and now mine updateDialogConfig

**Comment by NetanelBasal** [comment]
You can mock it

**Comment by va-stefanek** [comment]
@NetanelBasal I have added some basic test and documentation

File: projects/ngneat/dialog/src/lib/specs/dialog.component.spec.ts [hunk 1]
```diff
@@ -190,6 +190,12 @@ describe('DialogComponent', () => {
       dialogService.dialogs.push(Object.assign(spectator.component.dialogRef, { id: 'close-last-open-dialog' }));
     });
 
+    it('should properly update dialog config', () => {
```
Comment (NetanelBasal) [inline projects/ngneat/dialog/src/lib/specs/dialog.component.spec.ts:193]: can we please have more check? for example, let's update the width and change enableClose to false and see if it works

Comment (va-stefanek) [reply projects/ngneat/dialog/src/lib/specs/dialog.component.spec.ts]: I have added more testes let's me know whether they cover what expected

File: projects/ngneat/dialog/src/lib/specs/dialog.component.spec.ts [hunk 2]
```diff
@@ -199,6 +199,34 @@ describe('DialogComponent', () => {
...
+
+    it('should properly update size and id in config', () => {
+      expect(spectator.component.config.id).toBe('test');
+      expect(spectator.component.config.size).toBe('md');
+      spectator.component.updateDialogConfig({ size: 'lg', id: 'unique' });
+      expect(spectator.component.config.id).toBe('unique');
+      expect(spectator.component.config.size).toBe('lg');
+    });
+
+    it('should properly update width and enableClose in config', () => {
```
Comment (NetanelBasal) [inline projects/ngneat/dialog/src/lib/specs/dialog.component.spec.ts:221]: Can you check that the DOM is updated accordingly and not just the instance, please?

**Comment by shaharkazaz** [comment]
We appreciate the PR, I've added this feature (with a different API) as part of a refactor I've made.
It's available in v4.2.0, see the usage example under `DialogRef` [API](https://github.com/ngneat/dialog#dialogref-api)

**Pull request closed without merging**